include/cuda/std/detail/libcxx/include/__mdspan/trait_backports.hpp

```diff
@@ -0,0 +1,162 @@
+/*
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 2.0
+//              Copyright (2019) Sandia Corporation
+//
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef _LIBCUDACXX___MDSPAN_TRAIT_BACKPORTS_HPP
+#define _LIBCUDACXX___MDSPAN_TRAIT_BACKPORTS_HPP
+
+#include "macros.hpp"
+#include "config.hpp"
+
+#ifndef __cuda_std__
+#include <type_traits>
+#include <utility> // integer_sequence
+#endif
+
+//==============================================================================
+// <editor-fold desc="Variable template trait backports (e.g., is_void_v)"> {{{1
+
+#ifdef __MDSPAN_NEEDS_TRAIT_VARIABLE_TEMPLATE_BACKPORTS
+
+#if __MDSPAN_USE_VARIABLE_TEMPLATES
+namespace std {
+
+#define __MDSPAN_BACKPORT_TRAIT(__TRAIT) \
+  template <class... _Args> __MDSPAN_INLINE_VARIABLE constexpr auto __TRAIT##_v = __TRAIT<_Args...>::value;
+
+__MDSPAN_BACKPORT_TRAIT(is_assignable)
```

Comment:
Checking most of them I see them guarded by 
```cpp
#if _LIBCUDACXX_STD_VER > 11 && !defined(_LIBCUDACXX_HAS_NO_VARIABLE_TEMPLATES)
```

So I believe that is a remnant from a pre C++14 implementation?

Reply:
@miscco The macros wouldn't be original to the reference implementation, but there probably are some parts of the reference mdspan implementation that retain C++11 checks.  The reference mdspan implementation definitely requires C++14, though.

Reply:
This file if removed since we don't need C++11 support.